File: Easy/3340_Check_Balanced_String/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.

class Solution {
public:
    bool isBalanced(string num) {
        int evenSum = 0;
        int oddSum = 0;

        for (int i = 0; i < num.size(); i++) {
            int digit = num[i] - '0';
            if (i % 2 == 0) {
                evenSum += digit;
            } else {
                oddSum += digit;
            }
        }
        
        return evenSum == oddSum;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;
    
    // test cases 1
    cout << solution.isBalanced("1234") << endl; // Output: false

    // test cases 2
    cout << solution.isBalanced("24123") << endl;  // Output: true

    return 0;
}

File: Easy/3340_Check_Balanced_String/Solution.java


class Solution {
    public boolean isBalanced(String num) {
        int even_sum = 0;
        int odd_sum = 0;

        for (int i = 0; i < num.length(); i++) {
            int digit = num.charAt(i) - '0';

            if (i % 2 == 0) {
                even_sum += digit;
            } else {
                odd_sum += digit;
            }
        }

        return even_sum == odd_sum;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        System.out.println(solution.isBalanced("1234"));  // Output: False

        // test cases 2
        System.out.println(solution.isBalanced("24123"));  // Output: True
    }
}